<?xml version="1.0"?>
<robot name="ammr_main" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:property name="Silver" value="Gazebo/Silver" />
    <xacro:property name="Red" value="Gazebo/Red" />
    <xacro:property name="Blue" value="Gazebo/Blue" />
    <xacro:property name="White" value="Gazebo/White" />
    <xacro:property name="Green" value="Gazebo/Green" />
    <xacro:property name="Yellow" value="Gazebo/Yellow" />
    <xacro:property name="Orange" value="Gazebo/Orange" />

    <gazebo reference="caster_rear_right_link">
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
    </gazebo>
    <gazebo reference="caster_rear_left_link">
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
    </gazebo>
    <gazebo reference="caster_front_right_link">
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
    </gazebo>
    <gazebo reference="caster_front_left_link">
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
    </gazebo>

    <gazebo reference="base_link">
        <material>${White}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
        <gravity>true</gravity>
    </gazebo>

    <gazebo reference="Manipulator_J1_Joint">
        <material>${Red}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="Manipulator_J2_Joint">
        <material>${Green}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="Manipulator_J3_Joint">
        <material>${Blue}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="Manipulator_J4_Joint">
        <material>${Yellow}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="Manipulator_J5_Joint">
        <material>${Orange}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="Manipulator_J6_Joint">
        <material>${Silver}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="EE_R">
        <material>${Silver}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <gazebo reference="EE_L">
        <material>${Silver}</material>
        <mu1 value="0.1"/>
        <mu2 value="0.1"/>
        <self_collide>true</self_collide>
    </gazebo>

    <!--GAZEBO PLUGIN-->
    <gazebo>
        <plugin name="gazebo_ros2_arm_control" filename="libgazebo_ros2_control.so">
            <parameters>$(find ammr_moveit_config)/config/ros2_controllers.yaml</parameters>
            <!-- <ros>
                <remapping>/controller_manager/robot_description:=/robot_description</remapping>
            </ros> -->
        </plugin>
    </gazebo>

    <gazebo>
        <plugin name="gazebo_ros2_base_control" filename="libgazebo_ros_diff_drive.so" >
            <parameters>$(find ammr_moveit_config)/config/ros2_controllers.yaml</parameters>
            <left_joint>Left_Drive_wheel_Joint</left_joint>
            <right_joint>Right_Drive_wheel_Joint</right_joint>
            <wheel_separation>0.46</wheel_separation>
            <wheel_diameter>0.322</wheel_diameter>

            <max_wheel_torque>96.25</max_wheel_torque>
            <max_wheel_acceleration>1.5</max_wheel_acceleration>

            <frame_id>odom</frame_id>
            <child_frame_id>base_link</child_frame_id>

            <publish_odom>true</publish_odom>
            <publish_odom_tf>true</publish_odom_tf>
            <publish_wheel_tf>true</publish_wheel_tf>
            
        </plugin>
    </gazebo>

    <!-- <gazebo>
        <plugin
            filename="gz-sim-joint-state-publisher-system"
            name="gz::sim::systems::JointStatePublisher">
            <joint_name>Right_Drive_wheel_Joint</joint_name>
            <joint_name>Left_Drive_wheel_Joint</joint_name>
        </plugin>
    </gazebo> -->

</robot>